src/mpi/coll/bcast.c

```diff
@@ -1567,7 +1567,7 @@ int MPI_Bcast( void *buffer, int count, MPI_Datatype datatype, int root,
     /* ... body of routine ...  */
     
     mpi_errno = MPID_Bcast( buffer, count, datatype, root, comm_ptr, &errflag );
-    if (mpi_errno) goto fn_fail;
+    if (mpi_errno) MPIR_ERR_POP(mpi_errno);
```

Comment:
what's the reason for changing these `goto`s? I'm not saying they are wrong, just want to understand better.